components/b2b/QuickOrderTable/QuickOrderTable.spec.tsx

```diff
@@ -0,0 +1,139 @@
+
+    // assert
+    expect(productHeader).toBeVisible()
+    expect(fulfillmentMethodHeader).toBeVisible()
+    expect(quantityHeader).toBeVisible()
+    expect(priceHeader).toBeVisible()
+    expect(itemPriceHeader).toBeVisible()
+  })
+
+  it("should render table with 'No Products Added' caption, if cart is empty", () => {
+    // arrange
+    renderWithQueryClient(<Common {...Common.args} cartItems={[]} />)
+
+    // act
+    const cartItem = screen.queryAllByTestId(/product-item-component/i)
+
+    // assert
+
+    expect(cartItem.length).toBe(0)
+  })
+
+  it('should render table with cart items', () => {
+    // arrange
+    renderWithQueryClient(<Common {...Common.args} />)
+
+    // act
+    const productItems = screen.queryAllByTestId(/product-item-component/i)
+    const fulfillmentOptions = screen.queryAllByTestId(/fulfillment-options-component/i)
+    const quantitySelectors = screen.queryAllByTestId(/quantity-selector-component/i)
+    const prices = screen.queryAllByTestId(/price-component/i)
+
+    // assert
+    const count = Common.args?.cartItems?.length
+    expect(productItems.length).toBe(count)
+    expect(fulfillmentOptions.length).toBe(count)
+    expect(quantitySelectors.length).toBe(count)
+    expect(prices.length).toBe((count as number) * 2)
+  })
+
+  it('should call onQuantityUpdate when cart item quantity is changed', () => {
+    // arrange
+    const onQuantityUpdate = jest.fn()
+    renderWithQueryClient(<Common {...Common.args} onQuantityUpdate={onQuantityUpdate} />)
+
+    // act
+    const quantitySelector = screen.queryAllByTestId(/quantity-selector-component/i)
+    quantitySelector.forEach((selector, index) => {
+      const onQuantityUpdateButton = screen.getAllByTestId('onQuantityUpdate-button')
+      onQuantityUpdateButton[index].click()
+
+      const onIncreaseButton = screen.getAllByTestId('onIncrease-button')
+      onIncreaseButton[index].click()
+
+      const onDecreaseButton = screen.getAllByTestId('onDecrease-button')
+      onDecreaseButton[index].click()
+    })
+
+    // assert
+    expect(onQuantityUpdate).toHaveBeenCalledTimes((Common.args?.cartItems?.length as number) * 3)
+  })
+
+  it('should call onFulfillmentOptionChange when fulfillment option is changed', () => {
+    // arrange
+    const onFulfillmentOptionChange = jest.fn()
+    renderWithQueryClient(
+      <Common {...Common.args} onFulfillmentOptionChange={onFulfillmentOptionChange} />
+    )
+    const fulfillmentOptions = screen.queryAllByTestId(/fulfillment-options-component/i)
+
+    // act
+    fulfillmentOptions.forEach((option, index) => {
+      const onFulfillmentOptionChangeButton = screen.getAllByTestId(
+        'onFulfillmentOptionChange-button'
+      )
+      onFulfillmentOptionChangeButton[index].click()
+    })
+
+    // assert
+    expect(onFulfillmentOptionChange).toHaveBeenCalledTimes(
+      Common.args?.cartItems?.length as number
+    )
+  })
+})
```

Comment:
Missing Testcase
should call onStoreSetOrUpdate callback function when user selects store locator

Reply:
Resolved